draft-ietf-quic-transport.md

```diff
@@ -1698,10 +1698,14 @@ encrypted form in the token.
 It is unlikely that the client port number is the same on two different
 connections; validating the port is therefore unlikely to be successful.
 
-If the client has a token received in a NEW_TOKEN frame on a previous connection
-to what it believes to be the same server, it SHOULD include that value in the
-Token field of its Initial packet.  Including a token might allow the server to
-validate the client address without an additional round trip.
+A token received in a NEW_TOKEN frame is applicable against the servers that the
+connection is considered authoritative for (e.g., server names included in the
+certificate).  A client MUST only use a token that is applicable to the server
```

Comment:
```suggestion
certificate).  A client SHOULD only use a token that is applicable to the server
```

We don't need to prevent a client from using a token from one server on a completely different one.  That won't *work*, but the server needs to protect itself against outdated tokens and attack, so they shouldn't outright reject the connection.

Reply:
I thought you had an argument that we need a prohibition here; see https://github.com/quicwg/base-drafts/issues/3155#issuecomment-546789506.

Reply:
I don't remember making that argument, but I guess I could have said that.

The particular "attack" requires having a server privileged or capable enough to recognize a token from a different server with a fair degree of confidence.  For NEW_TOKEN, I can maybe see how that would happen though.  I'm OK with a MUST.

Reply:
@martinthomson FWIW I think the attack you were discussing was something different, that works even if the tokens were perfectly encrypted.

Consider the case where an attacker is an ISP and also controls example.com, which is hosted behind the large-scale load balancer hosted by a different entity (e.g., CDN). The attacker issues tokens when clients connect to example.com through the load balancer (IP address 192.0.2.1). Then, when the clients connect to different hostnames under the same load balancer (IP address 192.0.2.1), it can (as an ISP) see which exact clients went for which exact server-names.

I think you pointing out this attack has effectively killed https://datatracker.ietf.org/doc/draft-kazuho-quic-address-bound-token/, I appreciate that. But if it is the case that you think it's an attack we can ignore, I'm happy to revive the draft :-)